### PR TITLE
v12 lgg subtyping (12/N)

### PR DESCRIPTION
**Pull request by ewafula**
#### What scientific question is your analysis addressing?
Update LGG subtyping for v12  data release.

#### What was your approach?
- updated all outmoded gene symbols in both the module scripts and  gene/fusions input JSON files 
- updated module scripts
- included high-confidence methylation LGG subtypes from the histologies base file where subtypes are either to designated as `To be classified` or `wildtype`.
- regenerated new results with v12 pre-release datasets

#### What GitHub issue does your pull request address?
- https://github.com/PediatricOpenTargets/ticket-tracker/issues/488

#### Which areas should receive a particularly close look?
- Review all updated module scripts to check if the logic makes sense 
- Review the subtyping results

#### Is there anything that you want to discuss further?
NA

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
YES 

#### What types of results are included (e.g., table, figure)?
v12 LGG subtyping results  

#### What is your summary of the results?
v12 LGG subtyping results 

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

**Comment by jharenza** [comment]
Hi @ewafula can you check out what's going on with subtypes including FGFR, I see a few with duplicated FGFR:
Screenshot 2023-03-03 at 3 33 32 PM

**Comment by jharenza** [comment]
Can you also take `wildtype` out of the `molecular_subtype_methyl`, just because I don't think they subtype _all_ samples to as fine-grained subtypes we do, for eg- they may only have the molecular alterations for LGG, not SEGA.

File: analyses/molecular-subtyping-LGAT/01-subset-files-for-LGAT.Rmd [hunk 1]
```diff
@@ -313,13 +322,47 @@ lgat_subtypes
...
+                cohort != "TARGET",
+                dkfz_v12_methylation_subclass_score >= 0.8) %>% 
+  dplyr::mutate(molecular_subtype_methyl = 
+                  case_when(grepl("LGG_MYB", dkfz_v12_methylation_subclass) ~ "LGG, MYB/MYBL1",
+                            dkfz_v12_methylation_subclass == "SEGA" ~ "SEGA, wildtype",
+                            dkfz_v12_methylation_subclass == "A_IDH_LG" ~ "LGG, IDH",
+                            dkfz_v12_methylation_subclass == "PA_INF_FGFR" ~ "LGG, FGFR",
+                            dkfz_v12_methylation_subclass == "AG_MYB" ~ "LGG, MYB/MYBL1",
+                            dkfz_v12_methylation_subclass == "GG" ~ "GNG, wildtype")) %>% 
+  dplyr::select(Kids_First_Biospecimen_ID, Kids_First_Participant_ID, sample_id, molecular_subtype_methyl) %>%
```
Comment (jharenza) [inline analyses/molecular-subtyping-LGAT/01-subset-files-for-LGAT.Rmd]:
```suggestion
                            dkfz_v12_methylation_subclass == "SEGA" ~ "SEGA",
                            dkfz_v12_methylation_subclass == "A_IDH_LG" ~ "LGG, IDH",
                            dkfz_v12_methylation_subclass == "PA_INF_FGFR" ~ "LGG, FGFR",
                            dkfz_v12_methylation_subclass == "AG_MYB" ~ "LGG, MYB/MYBL1",
                            dkfz_v12_methylation_subclass == "GG" ~ "GNG, MAPK")) %>% 
  dplyr::select(Kids_First_Biospecimen_ID, Kids_First_Participant_ID, sample_id, molecular_subtype_methyl) %>%
```

Comment (ewafula) [reply analyses/molecular-subtyping-LGAT/01-subset-files-for-LGAT.Rmd]: Committed suggested change

> Hi @ewafula can you check out what's going on with subtypes including FGFR, I see a few with duplicated FGFR: Screenshot 2023-03-03 at 3 33 32 PM

@jharenza, now. It was a bug in the original code. Needed unique() to remove redundancy before applying toString().
https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/5fe7544ab3a95a9858d374c0c4504ab269993037/analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd#L303

**Review by zzgeng — APPROVED** [review]
Code works fine and logic is sound!

**Comment by jharenza** [comment]
@ewafula can you check 7316-20 - I am seeing methyl subtype FGFR, but our subtype KIAA1549--BRAF without FGFR.

**Comment by jharenza** [comment]
@ewafula can you also check these - since they are WT by ours and MAPK by methyl, we should take the methyl subtype.

```
Kids_First_Biospecimen_ID<chr> | sample_id<chr> | molecular_subtype<chr> | molecular_subtype_methyl<chr> |  
-- | -- | -- | -- | --
BS_1G3WYEN3 | 7316-193 | LGG, wildtype | GNG, MAPK |  
BS_258RRGDM | 7316-2195 | LGG, wildtype | GNG, MAPK |  
BS_QTEWYN0M | 7316-193 | LGG, wildtype | GNG, MAPK |  
BS_PMSZKZP7 | 7316-2195 | LGG, wildtype | GNG, MAPK |  
BS_PYSXW993 | 7316-193 | LGG, wildtype | GNG, MAPK |  
BS_QC00MR9J | 7316-2195 | LGG, wildtype | GNG, MAPK
```

**Comment by ewafula** [comment]
> @ewafula can you check 7316-20 - I am seeing methyl subtype FGFR, but our subtype KIAA1549--BRAF without FGFR.

@jharenza,  check 7316-20; it is not subtyped as FGFR. Modules list multiple subtypes in the molecular_subtype column if a sample has them.

**Comment by ewafula** [comment]
> @ewafula can you also check these - since they are WT by ours and MAPK by methyl, we should take the methyl subtype.
> 
> ```
> Kids_First_Biospecimen_ID<chr> | sample_id<chr> | molecular_subtype<chr> | molecular_subtype_methyl<chr> |  
> -- | -- | -- | -- | --
> BS_1G3WYEN3 | 7316-193 | LGG, wildtype | GNG, MAPK |  
> BS_258RRGDM | 7316-2195 | LGG, wildtype | GNG, MAPK |  
> BS_QTEWYN0M | 7316-193 | LGG, wildtype | GNG, MAPK |  
> BS_PMSZKZP7 | 7316-2195 | LGG, wildtype | GNG, MAPK |  
> BS_PYSXW993 | 7316-193 | LGG, wildtype | GNG, MAPK |  
> BS_QC00MR9J | 7316-2195 | LGG, wildtype | GNG, MAPK
> ```

@jharenza, updated the logic to replace wildtype in the molecular_subtype column if methyl has a subtype.

**Review by rjcorb — APPROVED** [review]
LGTM!

**Comment by ewafula** [comment]
@jharenza, you wanted this module updated using a `match _id`, which is already implemented in [script 04](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/1359366b9e253843859f977962d9a8932eca2fe1/analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd#L70-L92). The issue here is the `6 RNA` samples (`BS_40HCP3HB`,  `BS_MKGTXW8A`, `BS_MMEVDXGV`, `BS_P412CPXJ`, `BS_X5A5TNG1`, `BS_YV1E1G58`) duplicated in the final histologies file are each matched to  these`9  DNA` samples (`BS_0GVJJYWV`,  `BS_0MCY93CV`,  `BS_0NNV9QAF`,  `BS_1C5Y79HZ`,  `BS_A385FF70 (normal)`,  `BS_BYMV4DS7`,  `BS_STETBQHK`,  `BS_W8QVVDEJ`,  `BS_X6NW8R6K`) all from one patient (`PT_2JGWN6EA`) with the same sampe_id (`7316-8051`).  These 9 DNA LGG tumor samples are classified into 4 subtypes (`LGG, wildtype`,          `LGG, FGFR` , `LGG, NF1-somatic, FGFR`, `LGG, NF1-somatic`). This causes the 6 RNA samples to appear in the final histologies as duplicates because they are associated with 4 LGG subtypes. 

How do we use `match_id` to resolve? 
```
# histologies
histologies_df <- read_tsv("../../data/2023-04-11_histologies-base.tsv", 
                           guess_max = 100000) %>% 
  dplyr::select(Kids_First_Biospecimen_ID, Kids_First_Participant_ID, 
                sample_id, experimental_strategy, RNA_library) %>% 
  dplyr::mutate(match_id = 
                  paste(Kids_First_Participant_ID, sample_id, sep = "_"))

# DNA samples
dna <- histologies_df %>%
  dplyr::filter(experimental_strategy == "WGS" | 
                  experimental_strategy == "WXS" | 
                  experimental_strategy == "Targeted Sequencing", 
                is.na(RNA_library)) %>% 
  dplyr::rename(Kids_First_Biospecimen_ID_DNA = Kids_First_Biospecimen_ID) %>% 
  dplyr::select(-Kids_First_Participant_ID, -sample_id, 
                -experimental_strategy, -RNA_library )

# RNA samples
rna <- histologies_df %>%
  dplyr::filter(experimental_strategy == "RNA-Seq" | 
                  experimental_strategy == "Targeted Sequencing"
                & (!is.na(RNA_library) & 
                     experimental_strategy == "Targeted Sequencing")) %>% 
  dplyr::rename(Kids_First_Biospecimen_ID_RNA = Kids_First_Biospecimen_ID) %>% 
  dplyr::select(-Kids_First_Participant_ID, -sample_id, 
                -experimental_strategy, -RNA_library )


# merge DNA and RNA and filter for duplicated LGG 6 RNA samples in v12 histologies
dna_rna <- dna %>% 
  dplyr::inner_join(rna, by = "match_id") %>% 
  dplyr::filter(Kids_First_Biospecimen_ID_RNA %in% 
                  c("BS_40HCP3HB", "BS_MKGTXW8A", "BS_MMEVDXGV", 
                    "BS_P412CPXJ", "BS_X5A5TNG1", "BS_YV1E1G58")) %>% 
  dplyr::select(Kids_First_Biospecimen_ID_DNA, Kids_First_Biospecimen_ID_RNA, 
                match_id) %>% 
  dplyr::arrange(Kids_First_Biospecimen_ID_RNA)

> dna_rna %>% print(n=60)
# A tibble: 60 × 3
   Kids_First_Biospecimen_ID_DNA Kids_First_Biospecimen_ID_RNA match_id             
   <chr>                         <chr>                         <chr>                
 1 BS_0GVJJYWV                   BS_40HCP3HB                   PT_2JGWN6EA_7316-8051
 2 BS_0MCY93CV                   BS_40HCP3HB                   PT_2JGWN6EA_7316-8051
 3 BS_0NNV9QAF                   BS_40HCP3HB                   PT_2JGWN6EA_7316-8051
 4 BS_1C5Y79HZ                   BS_40HCP3HB                   PT_2JGWN6EA_7316-8051
 5 BS_A385FF70                   BS_40HCP3HB                   PT_2JGWN6EA_7316-8051
 6 BS_BYMV4DS7                   BS_40HCP3HB                   PT_2JGWN6EA_7316-8051
 7 BS_STETBQHK                   BS_40HCP3HB                   PT_2JGWN6EA_7316-8051
 8 BS_W8QVVDEJ                   BS_40HCP3HB                   PT_2JGWN6EA_7316-8051
 9 BS_X6NW8R6K                   BS_40HCP3HB                   PT_2JGWN6EA_7316-8051
10 BS_YAE6B9W7                   BS_40HCP3HB                   PT_2JGWN6EA_7316-8051
11 BS_0GVJJYWV                   BS_MKGTXW8A                   PT_2JGWN6EA_7316-8051
12 BS_0MCY93CV                   BS_MKGTXW8A                   PT_2JGWN6EA_7316-8051
13 BS_0NNV9QAF                   BS_MKGTXW8A                   PT_2JGWN6EA_7316-8051
14 BS_1C5Y79HZ                   BS_MKGTXW8A                   PT_2JGWN6EA_7316-8051
15 BS_A385FF70                   BS_MKGTXW8A                   PT_2JGWN6EA_7316-8051
16 BS_BYMV4DS7                   BS_MKGTXW8A                   PT_2JGWN6EA_7316-8051
17 BS_STETBQHK                   BS_MKGTXW8A                   PT_2JGWN6EA_7316-8051
18 BS_W8QVVDEJ                   BS_MKGTXW8A                   PT_2JGWN6EA_7316-8051
19 BS_X6NW8R6K                   BS_MKGTXW8A                   PT_2JGWN6EA_7316-8051
20 BS_YAE6B9W7                   BS_MKGTXW8A                   PT_2JGWN6EA_7316-8051
21 BS_0GVJJYWV                   BS_MMEVDXGV                   PT_2JGWN6EA_7316-8051
22 BS_0MCY93CV                   BS_MMEVDXGV                   PT_2JGWN6EA_7316-8051
23 BS_0NNV9QAF                   BS_MMEVDXGV                   PT_2JGWN6EA_7316-8051
24 BS_1C5Y79HZ                   BS_MMEVDXGV                   PT_2JGWN6EA_7316-8051
25 BS_A385FF70                   BS_MMEVDXGV                   PT_2JGWN6EA_7316-8051
26 BS_BYMV4DS7                   BS_MMEVDXGV                   PT_2JGWN6EA_7316-8051
27 BS_STETBQHK                   BS_MMEVDXGV                   PT_2JGWN6EA_7316-8051
28 BS_W8QVVDEJ                   BS_MMEVDXGV                   PT_2JGWN6EA_7316-8051
29 BS_X6NW8R6K                   BS_MMEVDXGV                   PT_2JGWN6EA_7316-8051
30 BS_YAE6B9W7                   BS_MMEVDXGV                   PT_2JGWN6EA_7316-8051
31 BS_0GVJJYWV                   BS_P412CPXJ                   PT_2JGWN6EA_7316-8051
32 BS_0MCY93CV                   BS_P412CPXJ                   PT_2JGWN6EA_7316-8051
33 BS_0NNV9QAF                   BS_P412CPXJ                   PT_2JGWN6EA_7316-8051
34 BS_1C5Y79HZ                   BS_P412CPXJ                   PT_2JGWN6EA_7316-8051
35 BS_A385FF70                   BS_P412CPXJ                   PT_2JGWN6EA_7316-8051
36 BS_BYMV4DS7                   BS_P412CPXJ                   PT_2JGWN6EA_7316-8051
37 BS_STETBQHK                   BS_P412CPXJ                   PT_2JGWN6EA_7316-8051
38 BS_W8QVVDEJ                   BS_P412CPXJ                   PT_2JGWN6EA_7316-8051
39 BS_X6NW8R6K                   BS_P412CPXJ                   PT_2JGWN6EA_7316-8051
40 BS_YAE6B9W7                   BS_P412CPXJ                   PT_2JGWN6EA_7316-8051
41 BS_0GVJJYWV                   BS_X5A5TNG1                   PT_2JGWN6EA_7316-8051
42 BS_0MCY93CV                   BS_X5A5TNG1                   PT_2JGWN6EA_7316-8051
43 BS_0NNV9QAF                   BS_X5A5TNG1                   PT_2JGWN6EA_7316-8051
44 BS_1C5Y79HZ                   BS_X5A5TNG1                   PT_2JGWN6EA_7316-8051
45 BS_A385FF70                   BS_X5A5TNG1                   PT_2JGWN6EA_7316-8051
46 BS_BYMV4DS7                   BS_X5A5TNG1                   PT_2JGWN6EA_7316-8051
47 BS_STETBQHK                   BS_X5A5TNG1                   PT_2JGWN6EA_7316-8051
48 BS_W8QVVDEJ                   BS_X5A5TNG1                   PT_2JGWN6EA_7316-8051
49 BS_X6NW8R6K                   BS_X5A5TNG1                   PT_2JGWN6EA_7316-8051
50 BS_YAE6B9W7                   BS_X5A5TNG1                   PT_2JGWN6EA_7316-8051
51 BS_0GVJJYWV                   BS_YV1E1G58                   PT_2JGWN6EA_7316-8051
52 BS_0MCY93CV                   BS_YV1E1G58                   PT_2JGWN6EA_7316-8051
53 BS_0NNV9QAF                   BS_YV1E1G58                   PT_2JGWN6EA_7316-8051
54 BS_1C5Y79HZ                   BS_YV1E1G58                   PT_2JGWN6EA_7316-8051
55 BS_A385FF70                   BS_YV1E1G58                   PT_2JGWN6EA_7316-8051
56 BS_BYMV4DS7                   BS_YV1E1G58                   PT_2JGWN6EA_7316-8051
57 BS_STETBQHK                   BS_YV1E1G58                   PT_2JGWN6EA_7316-8051
58 BS_W8QVVDEJ                   BS_YV1E1G58                   PT_2JGWN6EA_7316-8051
59 BS_X6NW8R6K                   BS_YV1E1G58                   PT_2JGWN6EA_7316-8051
60 BS_YAE6B9W7                   BS_YV1E1G58                   PT_2JGWN6EA_7316-8051
```

```
read_tsv("lgat_subtyping.tsv") %>% 
  dplyr::filter(Kids_First_Biospecimen_ID_DNA %in%
                  c("BS_0GVJJYWV", "BS_0MCY93CV", "BS_0NNV9QAF", 
                    "BS_1C5Y79HZ", "BS_BYMV4DS7", "BS_STETBQHK", 
                    "BS_W8QVVDEJ", "BS_X6NW8R6K")) %>% 
  dplyr::pull(molecular_subtype) %>% 
  unique()
[1] "LGG, wildtype"    "LGG, FGFR"    "LGG, NF1-somatic, FGFR"    "LGG, NF1-somatic"
```

File: analyses/molecular-subtyping-LGAT/01-subset-files-for-LGAT.Rmd [hunk 2]
```diff
@@ -74,7 +74,7 @@ clinical <- read_tsv(file.path(root_dir,
 ```
 
 ### SNV for subtyping 
- - We will use the `snv-consensus-plus-hotspots.maf.tsv.gz` which contained consensus calling plus hotspot rescued for SNV calling - detilaed in [here](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/snv-callers#snv-caller-comparison-analysis))
+ - We will use the `snv-consensus-plus-hotspots.maf.tsv.gz` which contained consensus calling plus hotspot rescued for SNV calling - detialed in [here](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/snv-callers#snv-caller-comparison-analysis)). Also include the DGD MAF, `snv-dgd.maf.tsv.gz` for OpenPedCan.
```
Comment (jharenza) [inline analyses/molecular-subtyping-LGAT/01-subset-files-for-LGAT.Rmd]:
```suggestion
 - We will use the `snv-consensus-plus-hotspots.maf.tsv.gz` which contained consensus calling plus hotspot rescued for SNV calling - detailed in [here](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/snv-callers#snv-caller-comparison-analysis)). Also include the DGD MAF, `snv-dgd.maf.tsv.gz` for OpenPedCan.
```

File: analyses/molecular-subtyping-LGAT/01-subset-files-for-LGAT.Rmd [hunk 3]
```diff
@@ -313,13 +322,47 @@ lgat_subtypes
...
+                composition == "Solid Tissue",
+                cohort != "TARGET",
+                dkfz_v12_methylation_subclass_score >= 0.8) %>% 
+  dplyr::mutate(molecular_subtype_methyl = 
+                  case_when(grepl("LGG_MYB", dkfz_v12_methylation_subclass) ~ "LGG, MYB/MYBL1",
+                            dkfz_v12_methylation_subclass == "SEGA" ~ "SEGA, To be classified",
+                            dkfz_v12_methylation_subclass == "A_IDH_LG" ~ "LGG, IDH",
+                            dkfz_v12_methylation_subclass == "PA_INF_FGFR" ~ "LGG, FGFR",
+                            dkfz_v12_methylation_subclass == "AG_MYB" ~ "LGG, MYB/MYBL1",
+                            dkfz_v12_methylation_subclass == "GG" ~ "GNG, MAPK")) %>%
```
Comment (jharenza) [inline analyses/molecular-subtyping-LGAT/01-subset-files-for-LGAT.Rmd]: Can you also add these:

Abbrevation | molecular_subtype_methyl
-- | --
A_IDH_LG | LGG, IDH
AG_MYB | LGG, MYB/MYBL1
DGONC | GNT
DIG_DIA | GNT
DLGNT_1 | GNT, MAPK
DLGNT_2 | GNT, MAPK
GG | GNG, MAPK
LGG_MYB_B | LGG, MYB/MYBL1
LGG_MYB_C | LGG, MYB/MYBL1
LGG_MYB_D | LGG, MYB/MYBL1
PA_INF_FGFR | LGG, FGFR
DNET | GNT
GNT_A | GNT
PA_CORT | GNT, MAPK
PA_INF | LGG, FGFR
PA_INF_FGFR | LGG, FGFR
PA_MID | LGG, MAPK
PLNTY | LGG, MAPK
RGNT | GNT, FGFR
PXA | PXA

**Pull request merged**